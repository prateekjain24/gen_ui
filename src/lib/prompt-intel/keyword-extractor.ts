import { TOOL_DEFINITIONS } from './tool-definitions';
import type {
  CopyTone,
  ComplianceTag,
  PromptSignalMetadata,
  PromptSignalValue,
  PromptSignalsPartial,
  TeamSizeBracket,
  ToolIdentifier,
} from './types';

import { createDebugger } from '@/lib/utils/debug';

const debug = createDebugger('PromptIntel:KeywordExtractor');

const KEYWORD_CONFIDENCE = 1;

interface NormalizedPrompt {
  original: string;
  lower: string;
  sanitized: string;
}

const TOOL_KEYWORDS: Record<ToolIdentifier, string[]> = TOOL_DEFINITIONS.reduce(
  (accumulator, definition) => {
    accumulator[definition.id] = definition.keywords;
    return accumulator;
  },
  {} as Record<ToolIdentifier, string[]>
);

const COMPLIANCE_KEYWORDS: Record<Exclude<ComplianceTag, 'other'>, string[]> = {
  SOC2: ['soc2', 'soc 2'],
  HIPAA: ['hipaa'],
  ISO27001: ['iso27001', 'iso 27001'],
  GDPR: ['gdpr', 'privacy law', 'privacy regulation'],
  SOX: ['sox', 'sarbanes-oxley', 'sarbanes oxley'],
  audit: ['audit ready', 'audit-ready', 'audit'],
  'regulated-industry': ['regulated industry', 'highly regulated', 'regulated market'],
};

const GENERAL_COMPLIANCE_KEYWORDS = ['compliance', 'regulatory', 'regulation'];

const TONE_KEYWORDS: Array<{ tone: CopyTone; keywords: string[] }> = [
  {
    tone: 'fast-paced',
    keywords: ['fast paced', 'fast-paced', 'punchy', 'energetic', 'snappy', 'quick turnaround'],
  },
  {
    tone: 'meticulous',
    keywords: ['meticulous', 'detailed', 'thorough', 'buttoned up', 'buttoned-up', 'compliance focused'],
  },
  {
    tone: 'trusted-advisor',
    keywords: ['trusted advisor', 'trusted-advisor', 'advisory', 'consultative', 'guidance'],
  },
  {
    tone: 'onboarding',
    keywords: ['onboarding', 'welcome experience', 'getting started'],
  },
  {
    tone: 'migration',
    keywords: ['migration', 'cutover', 'data move'],
  },
];

const SOLO_KEYWORDS = ['solo', 'just me', 'individual contributor', 'one person', 'single founder'];
const SMALL_TEAM_KEYWORDS = ['small team', 'under 10', 'less than 10', 'tiny team'];
const MID_TEAM_KEYWORDS = ['mid-sized team', 'mid size team', 'mid team', 'growth team'];
const LARGE_TEAM_KEYWORDS = ['large team', 'enterprise team', 'big team', 'over 25'];

export function extractSignalsFromKeywords(prompt: string): PromptSignalsPartial {
  const partial: PromptSignalsPartial = {};

  if (!prompt || !prompt.trim()) {
    debug('No prompt provided to keyword extractor');
    return partial;
  }

  const normalized = normalizePrompt(prompt);

  const teamSize = detectTeamSize(normalized);
  if (teamSize) {
    partial.teamSizeBracket = teamSize;
  }

  const tools = detectTools(normalized);
  if (tools) {
    partial.tools = tools;
  }

  const compliance = detectCompliance(normalized);
  if (compliance) {
    partial.complianceTags = compliance;
  }

  const tone = detectTone(normalized);
  if (tone) {
    partial.copyTone = tone;
  }

  if (Object.keys(partial).length === 0) {
    debug('No keyword signals detected for prompt snippet: "%s"', normalized.original.slice(0, 80));
  }

  return partial;
}

function normalizePrompt(prompt: string): NormalizedPrompt {
  const lower = prompt.toLowerCase();
  const sanitized = lower.replace(/[^a-z0-9\s]/g, ' ').replace(/\s+/g, ' ').trim();

  return {
    original: prompt,
    lower,
    sanitized,
  };
}

function detectTeamSize(normalized: NormalizedPrompt): PromptSignalValue<TeamSizeBracket> | undefined {
  const { lower, original } = normalized;

  for (const phrase of SOLO_KEYWORDS) {
    if (lower.includes(phrase)) {
      return signalValue('solo', `keyword: ${phrase}`);
    }
  }

  const rangeMatches = collectRangeMatches(original);

  const explicitRegex = /(\d{1,3})(?:\s*-\s*(?=person|people|member|team))?\s*(?:person|people|member|team)s?\b/gi;
  let explicitMatch: RegExpExecArray | null;
  while ((explicitMatch = explicitRegex.exec(original)) !== null) {
    if (isWithinRange(rangeMatches, explicitMatch.index)) {
      continue;
    }

    const value = Number(explicitMatch[1]);
    const bracket = bracketForHeadcount(value);
    if (bracket) {
      return signalValue(bracket, `matched explicit headcount "${explicitMatch[0]}"`);
    }
  }

  const plusRegex = /(\d{1,3})\s*\+/gi;
  let plusMatch: RegExpExecArray | null;
  while ((plusMatch = plusRegex.exec(original)) !== null) {
    if (isWithinRange(rangeMatches, plusMatch.index)) {
      continue;
    }

    const value = Number(plusMatch[1]);
    const bracket = bracketForHeadcount(value < 25 ? value : 25);
    if (bracket) {
      return signalValue(bracket, `matched plus headcount "${plusMatch[0]}"`);
    }
  }

  if (rangeMatches.length > 0) {
    const [firstRange] = rangeMatches;
    const bracket = bracketForHeadcount(Math.round((firstRange.min + firstRange.max) / 2));
    if (bracket) {
      return signalValue(bracket, `matched range "${firstRange.raw}"`);
    }
  }

  for (const phrase of SMALL_TEAM_KEYWORDS) {
    if (lower.includes(phrase)) {
      return signalValue('1-9', `keyword: ${phrase}`);
    }
  }

  for (const phrase of MID_TEAM_KEYWORDS) {
    if (lower.includes(phrase)) {
      return signalValue('10-24', `keyword: ${phrase}`);
    }
  }

  for (const phrase of LARGE_TEAM_KEYWORDS) {
    if (lower.includes(phrase)) {
      return signalValue('25+', `keyword: ${phrase}`);
    }
  }

  const looseNumberRegex = /(\d{1,3})/gi;
  let looseMatch: RegExpExecArray | null;
  while ((looseMatch = looseNumberRegex.exec(original)) !== null) {
    if (isWithinRange(rangeMatches, looseMatch.index)) {
      continue;
    }

    const value = Number(looseMatch[1]);
    const bracket = bracketForHeadcount(value);
    if (bracket) {
      return signalValue(bracket, `matched numeric headcount "${looseMatch[0]}"`);
    }
  }

  return undefined;
}

function detectTools(normalized: NormalizedPrompt): PromptSignalValue<ToolIdentifier[]> | undefined {
  const matchedTools: ToolIdentifier[] = [];
  const notes: string[] = [];

  for (const [tool, keywords] of Object.entries(TOOL_KEYWORDS) as Array<[
    ToolIdentifier,
    string[],
  ]>) {
    if (tool === 'Other') {
      continue;
    }

    const hit = keywords.find((keyword) => normalized.sanitized.includes(keyword));
    if (hit) {
      matchedTools.push(tool);
      notes.push(`${tool} (${hit})`);
    }
  }

  if (matchedTools.length === 0) {
    return undefined;
  }

  const uniqueTools = Array.from(new Set(matchedTools));

  return signalValue(uniqueTools, `keywords: ${notes.join(', ')}`);
}

function detectCompliance(normalized: NormalizedPrompt): PromptSignalValue<ComplianceTag[]> | undefined {
  const matches = new Set<ComplianceTag>();
  const notes: string[] = [];

  for (const [tag, keywords] of Object.entries(COMPLIANCE_KEYWORDS) as Array<[
    Exclude<ComplianceTag, 'other'>,
    string[],
  ]>) {
    const hit = keywords.find((keyword) => normalized.sanitized.includes(keyword));
    if (hit) {
      matches.add(tag);
      notes.push(`${tag} (${hit})`);
    }
  }

  if (matches.size === 0) {
    const generalHit = GENERAL_COMPLIANCE_KEYWORDS.find((keyword) =>
      normalized.sanitized.includes(keyword),
    );
    if (generalHit) {
      matches.add('other');
      notes.push(`other (${generalHit})`);
    }
  }

  if (matches.size === 0) {
    return undefined;
  }

  return signalValue(Array.from(matches), `keywords: ${notes.join(', ')}`);
}

function detectTone(normalized: NormalizedPrompt): PromptSignalValue<CopyTone> | undefined {
  for (const { tone, keywords } of TONE_KEYWORDS) {
    const hit = keywords.find((keyword) => normalized.sanitized.includes(keyword));
    if (hit) {
      return signalValue(tone, `keyword: ${hit}`);
    }
  }

  return undefined;
}

function bracketForHeadcount(value: number): TeamSizeBracket | undefined {
  if (Number.isNaN(value) || value <= 0) {
    return undefined;
  }

  if (value <= 1) {
    return 'solo';
  }

  if (value <= 9) {
    return '1-9';
  }

  if (value <= 24) {
    return '10-24';
  }

  return '25+';
}

interface RangeMatch {
  start: number;
  end: number;
  min: number;
  max: number;
  raw: string;
}

function collectRangeMatches(text: string): RangeMatch[] {
  const regex = /(\d{1,3})\s*(?:-|to)\s*(\d{1,3})/gi;
  const matches: RangeMatch[] = [];
  let match: RegExpExecArray | null;

  while ((match = regex.exec(text)) !== null) {
    matches.push({
      start: match.index,
      end: regex.lastIndex,
      min: Number(match[1]),
      max: Number(match[2]),
      raw: match[0],
    });
  }

  return matches;
}

function isWithinRange(ranges: RangeMatch[], index: number): boolean {
  return ranges.some((range) => index >= range.start && index < range.end);
}

function signalValue<T>(value: T, notes?: string): PromptSignalValue<T> {
  return {
    value,
    metadata: makeMetadata(notes),
  };
}

function makeMetadata(notes?: string): PromptSignalMetadata {
  return {
    source: 'keyword',
    confidence: KEYWORD_CONFIDENCE,
    ...(notes ? { notes } : {}),
  };
}
