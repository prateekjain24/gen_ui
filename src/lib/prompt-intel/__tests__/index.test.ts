import fixtures from '../__fixtures__';
import { buildPromptSignals, summarizePromptSignals } from '../index';
import { extractSignalsFromKeywords } from '../keyword-extractor';

jest.mock('../llm-parser', () => ({
  fetchSignalsFromLLM: jest.fn(),
}));

jest.mock('@/lib/utils/debug', () => {
  const actual = jest.requireActual('@/lib/utils/debug');
  return {
    ...actual,
    createDebugger: jest.fn(() => jest.fn()),
  };
});

const { fetchSignalsFromLLM } = jest.requireMock('../llm-parser');

describe('buildPromptSignals', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('returns default signals for empty prompt', async () => {
    const result = await buildPromptSignals('   ');

    expect(result.teamSizeBracket.value).toBe('unknown');
    expect(result.teamSizeBracket.metadata.source).toBe('merge');
    expect(fetchSignalsFromLLM).not.toHaveBeenCalled();
  });

  it('uses keyword signals when LLM returns nothing', async () => {
    fetchSignalsFromLLM.mockResolvedValue({});

    const prompt =
      'Need a punchy migration plan for our 40-person healthcare onboarding team. Must stay HIPAA compliant and integrate Zendesk.';

    const keywordSignals = extractSignalsFromKeywords(prompt);
    const result = await buildPromptSignals(prompt);

    expect(fetchSignalsFromLLM).toHaveBeenCalledTimes(1);
    expect(result.teamSizeBracket.value).toBe(keywordSignals.teamSizeBracket?.value);
    expect(result.teamSizeBracket.metadata.source).toBe('keyword');
    expect(result.copyTone.value).toBe('fast-paced');
    expect(result.copyTone.metadata.source).toBe('keyword');
  });

  it('prefers LLM value when confidence exceeds threshold', async () => {
    fetchSignalsFromLLM.mockResolvedValue({
      teamSizeBracket: {
        value: '10-24',
        metadata: { source: 'llm', confidence: 0.92 },
      },
      copyTone: {
        value: 'meticulous',
        metadata: { source: 'llm', confidence: 0.88 },
      },
    });

    const prompt =
      'We are a 40-person operations crew rolling out Salesforce. Need punchy copy and Slack automation callouts.';

    const result = await buildPromptSignals(prompt);

    expect(result.teamSizeBracket.value).toBe('10-24');
    expect(result.teamSizeBracket.metadata.source).toBe('llm');
    expect(result.copyTone.value).toBe('meticulous');
    expect(result.copyTone.metadata.source).toBe('llm');
  });

  it('retains keyword value when LLM confidence is below threshold', async () => {
    fetchSignalsFromLLM.mockResolvedValue({
      tools: {
        value: ['Notion'],
        metadata: { source: 'llm', confidence: 0.5 },
      },
    });

    const prompt = 'Integrate Slack and Jira for our 12 person team.';

    const result = await buildPromptSignals(prompt);

    expect(result.tools.value).toEqual(['Slack', 'Jira']);
    expect(result.tools.metadata.source).toBe('keyword');
  });

  it('merges matching signals and reports combined summary', async () => {
    fetchSignalsFromLLM.mockResolvedValue({
      complianceTags: {
        value: ['HIPAA'],
        metadata: { source: 'llm', confidence: 0.7 },
      },
    });

    const prompt = 'HIPAA compliant workflows for healthcare onboarding.';
    const result = await buildPromptSignals(prompt);

    expect(result.complianceTags.metadata.source).toBe('merge');
    expect(result.complianceTags.metadata.confidence).toBeGreaterThanOrEqual(0.7);

    const summary = summarizePromptSignals(result);
    const complianceSummary = summary.find(item => item.key === 'complianceTags');
    expect(complianceSummary?.source).toBe('merge');
    expect(complianceSummary?.confidence).toBeGreaterThan(0.6);
  });
});

describe('prompt fixtures', () => {
  it.each(fixtures)("produces stable signals for %s", async fixture => {
    fetchSignalsFromLLM.mockReset();
    fetchSignalsFromLLM.mockResolvedValueOnce(fixture.llm ?? {});

    const result = await buildPromptSignals(fixture.prompt);
    const serializable = JSON.parse(JSON.stringify(result));

    expect(serializable).toMatchSnapshot(fixture.id);
  });
});
