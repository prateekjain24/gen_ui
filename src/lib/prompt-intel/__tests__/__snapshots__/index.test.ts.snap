// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`prompt fixtures produces stable signals for {\\n  id: 'global-saas',\\n  prompt: "We're scaling a 22-person RevOps org across North America and APAC. Legal and finance leads must approve every launch. Need Salesforce and Slack automation, SOC2 ready messaging, and onboarding tone that's meticulous but still upbeat.",\\n  llm: [Object]\\n}: global-saas 1`] = `
{
  "approvalChainDepth": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "unknown",
  },
  "complianceTags": {
    "metadata": {
      "confidence": 1,
      "notes": "keywords: SOC2 (soc2)",
      "source": "keyword",
    },
    "value": [
      "SOC2",
    ],
  },
  "constraints": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": {},
  },
  "copyTone": {
    "metadata": {
      "confidence": 1,
      "notes": "keyword: meticulous",
      "source": "keyword",
    },
    "value": "meticulous",
  },
  "decisionMakers": {
    "metadata": {
      "confidence": 0.86,
      "source": "llm",
    },
    "value": [
      {
        "isPrimary": true,
        "role": "Head of Legal",
        "seniority": "director+",
      },
      {
        "isPrimary": true,
        "role": "VP Finance",
        "seniority": "director+",
      },
    ],
  },
  "industry": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "other",
  },
  "integrationCriticality": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "unspecified",
  },
  "operatingRegion": {
    "metadata": {
      "confidence": 0.8,
      "source": "llm",
    },
    "value": "global",
  },
  "primaryObjective": {
    "metadata": {
      "confidence": 0.74,
      "source": "llm",
    },
    "value": "scale",
  },
  "teamSizeBracket": {
    "metadata": {
      "confidence": 1,
      "notes": "matched explicit headcount "22-person"",
      "source": "keyword",
    },
    "value": "10-24",
  },
  "tools": {
    "metadata": {
      "confidence": 1,
      "notes": "keywords: Slack (slack), Salesforce (salesforce)",
      "source": "keyword",
    },
    "value": [
      "Slack",
      "Salesforce",
    ],
  },
}
`;

exports[`prompt fixtures produces stable signals for {\\n  id: 'multilingual-expansion',\\n  prompt: 'Somos un equipo de 8 personas en Madrid preparando migración a Zendesk y Slack. Urgente: listo antes del onboarding de partners en EMEA. Necesitamos tono energético y notas sobre presupuesto ajustado.',\\n  llm: [Object]\\n}: multilingual-expansion 1`] = `
{
  "approvalChainDepth": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "unknown",
  },
  "complianceTags": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": [],
  },
  "constraints": {
    "metadata": {
      "confidence": 0.72,
      "source": "llm",
    },
    "value": {
      "budget": "tight",
      "notes": "Onboarding partners in EMEA",
      "timeline": "rush",
    },
  },
  "copyTone": {
    "metadata": {
      "confidence": 0.76,
      "notes": "LLM override of keyword value | keyword: onboarding",
      "source": "llm",
    },
    "value": "fast-paced",
  },
  "decisionMakers": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": [],
  },
  "industry": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "other",
  },
  "integrationCriticality": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "unspecified",
  },
  "operatingRegion": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "unspecified",
  },
  "primaryObjective": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "other",
  },
  "teamSizeBracket": {
    "metadata": {
      "confidence": 1,
      "notes": "Keyword and LLM agreement | matched numeric headcount "8"",
      "source": "merge",
    },
    "value": "1-9",
  },
  "tools": {
    "metadata": {
      "confidence": 1,
      "notes": "keywords: Slack (slack), Zendesk (zendesk)",
      "source": "keyword",
    },
    "value": [
      "Slack",
      "Zendesk",
    ],
  },
}
`;

exports[`prompt fixtures produces stable signals for {\\n  id: 'partial-sentence-brief',\\n  prompt: '40-person onboarding crew. migrating support flows to ServiceNow. HIPAA, audit logging mandatory. migration tone = trusted advisor.',\\n  llm: [Object]\\n}: partial-sentence-brief 1`] = `
{
  "approvalChainDepth": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "unknown",
  },
  "complianceTags": {
    "metadata": {
      "confidence": 1,
      "notes": "keywords: HIPAA (hipaa), audit (audit)",
      "source": "keyword",
    },
    "value": [
      "HIPAA",
      "audit",
    ],
  },
  "constraints": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": {},
  },
  "copyTone": {
    "metadata": {
      "confidence": 1,
      "notes": "keyword: trusted advisor",
      "source": "keyword",
    },
    "value": "trusted-advisor",
  },
  "decisionMakers": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": [],
  },
  "industry": {
    "metadata": {
      "confidence": 0.82,
      "source": "llm",
    },
    "value": "healthcare",
  },
  "integrationCriticality": {
    "metadata": {
      "confidence": 0.7,
      "source": "llm",
    },
    "value": "must-have",
  },
  "operatingRegion": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "unspecified",
  },
  "primaryObjective": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "other",
  },
  "teamSizeBracket": {
    "metadata": {
      "confidence": 1,
      "notes": "matched explicit headcount "40-person"",
      "source": "keyword",
    },
    "value": "25+",
  },
  "tools": {
    "metadata": {
      "confidence": 1,
      "notes": "keywords: ServiceNow (servicenow)",
      "source": "keyword",
    },
    "value": [
      "ServiceNow",
    ],
  },
}
`;

exports[`prompt fixtures produces stable signals for {\\n  id: 'regulated-healthcare',\\n  prompt: 'Need meticulous patient onboarding for a 65-person clinical ops team. Must satisfy HIPAA, ISO27001, and internal audit checkpoints. Tools in scope: Notion for templates, Slack alerts, and Zendesk escalations. Stakeholder: Chief Compliance Officer signs off solo.',\\n  llm: [Object]\\n}: regulated-healthcare 1`] = `
{
  "approvalChainDepth": {
    "metadata": {
      "confidence": 0.68,
      "source": "llm",
    },
    "value": "single",
  },
  "complianceTags": {
    "metadata": {
      "confidence": 1,
      "notes": "keywords: HIPAA (hipaa), ISO27001 (iso27001), audit (audit)",
      "source": "keyword",
    },
    "value": [
      "HIPAA",
      "ISO27001",
      "audit",
    ],
  },
  "constraints": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": {},
  },
  "copyTone": {
    "metadata": {
      "confidence": 1,
      "notes": "keyword: meticulous",
      "source": "keyword",
    },
    "value": "meticulous",
  },
  "decisionMakers": {
    "metadata": {
      "confidence": 0.83,
      "source": "llm",
    },
    "value": [
      {
        "isPrimary": true,
        "role": "Chief Compliance Officer",
        "seniority": "director+",
      },
    ],
  },
  "industry": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "other",
  },
  "integrationCriticality": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "unspecified",
  },
  "operatingRegion": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "unspecified",
  },
  "primaryObjective": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "other",
  },
  "teamSizeBracket": {
    "metadata": {
      "confidence": 1,
      "notes": "keyword: solo",
      "source": "keyword",
    },
    "value": "solo",
  },
  "tools": {
    "metadata": {
      "confidence": 1,
      "notes": "keywords: Slack (slack), Notion (notion), Zendesk (zendesk)",
      "source": "keyword",
    },
    "value": [
      "Slack",
      "Notion",
      "Zendesk",
    ],
  },
}
`;

exports[`prompt fixtures produces stable signals for {\\n  id: 'tooling-overlap',\\n  prompt: 'Solo founder sprinting on a pilot. Mentions: Slack standups, slack huddles, Jira automations, jira roadmaps, Notion docs. Needs calm onboarding tone, nothing salesy.',\\n  llm: [Object]\\n}: tooling-overlap 1`] = `
{
  "approvalChainDepth": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "unknown",
  },
  "complianceTags": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": [],
  },
  "constraints": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": {},
  },
  "copyTone": {
    "metadata": {
      "confidence": 1,
      "notes": "Keyword preferred over lower-confidence LLM | keyword: onboarding",
      "source": "keyword",
    },
    "value": "onboarding",
  },
  "decisionMakers": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": [],
  },
  "industry": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "other",
  },
  "integrationCriticality": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "unspecified",
  },
  "operatingRegion": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "unspecified",
  },
  "primaryObjective": {
    "metadata": {
      "confidence": 0,
      "source": "merge",
    },
    "value": "other",
  },
  "teamSizeBracket": {
    "metadata": {
      "confidence": 1,
      "notes": "keyword: solo",
      "source": "keyword",
    },
    "value": "solo",
  },
  "tools": {
    "metadata": {
      "confidence": 1,
      "notes": "keywords: Slack (slack), Jira (jira), Notion (notion)",
      "source": "keyword",
    },
    "value": [
      "Slack",
      "Jira",
      "Notion",
    ],
  },
}
`;
